src/components/common/AccessBlocker.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import { getUserProfile } from 'console/state/auth/selectors';
+import { isNormandyAdminAvailable } from 'console/state/network/selectors';
+import AuthenticationAlert from 'console/components/common/AuthenticationAlert';
+import VPNAlert from 'console/components/common/VPNAlert';
+
+function AccessBlocker(ProtectedComponent) {
+  @connect(
+    state => {
+      return { userProfile: getUserProfile(state), vpnAvailable: isNormandyAdminAvailable(state) };
+    },
+    {},
+  )
+  class AccessBlockerInner extends React.Component {
+    static propTypes = {
+      // XXX How do you make this `null | instanceOf(Map)`??
```

Comment:
you could do something along the lines of:
```js
PropTypes.oneOf([
  PropTypes.instanceOf(Map),
  PropTypes.instanceOf(null),
]),
```

although i always thought if it didn't have `.isRequired` then the `null` should be fine.

Reply:
Maybe there's a difference between absent and `null` and PropTypes does something like:
```javascript
if (value !== undefined) {
  check(value, expectedType)
}
```

More importantly, I fear there's some lack of understanding here. 

If you put this same `@connect(state => { return {userProfile: getUserProfile(state)} ...)` on the `CreateRecipePage` component instead, then it's never null. ...if you're logged in. 

But when it's here on the `AccessBlockerInner` it's `null` briefly and whence the auth-js check is complete it becomes an actual `Map` object.

Reply:
@rehandalal Would you mind taking a look and try to find out why, when logged in, the `userProfile` is never `null` when used on page components like `CreateRecipePage` but *can be* on the `AccessBlocker` wrapper component.